PageViews/PageViews.js

```diff
@@ -0,0 +1,295 @@
+				</Cell>
+			);
+		},
+		nextNavigationButton: ({index, onNextClick, totalIndex}) => {
+			const isNextButtonVisible = index < totalIndex - 1;
+
+			return (
+				<Cell className={css.navButton} shrink>
+					{isNextButtonVisible ? <Button aria-label={$L('Next')} icon="arrowlargeright" iconFlip="auto" id="NextNavButton" onClick={onNextClick} /> : null}
+				</Cell>
+			);
+		},
+		steps: ({index, onNextClick, onPrevClick, pageIndicatorType, totalIndex}) => {
+			const isPrevButtonVisible = index !== 0;
+			const isNextButtonVisible = index < totalIndex - 1;
+			return (
+				pageIndicatorType === 'dot' ?
+					<Column align="center center" className={css.stepsArea}>
+						<Steps
+							current={index + 1}
+							pastIcon={'circle'}
+							currentIcon={'circle'}
+							futureIcon={'circle'}
+							layout="quickGuidePanels"
+							total={totalIndex}
+						/>
+					</Column> :
+					<Column align="center center" className={css.stepsArea}>
+						<Row className={css.steps}>
+							<Cell className={css.navButton} shrink>
+								{isPrevButtonVisible ? <Button aria-label={$L('Previous')} icon="arrowlargeleft" iconFlip="auto" id="PrevNavButton" onClick={onPrevClick} /> : null}
+							</Cell>
+							<Cell shrink>{index + 1} / {totalIndex}</Cell>
+							<Cell className={css.navButton} shrink>
+								{isNextButtonVisible ? <Button aria-label={$L('Next')} icon="arrowlargeright" iconFlip="auto" id="NextNavButton" onClick={onNextClick} /> : null}
+							</Cell>
+						</Row>
+					</Column>
+			);
+		}
+	},
+	render: ({
+		arranger,
+		children,
+		index,
+		nextNavigationButton,
+		noAnimation,
+		onTransition,
+		onWillTransition,
+		pageIndicatorType,
+		prevNavigationButton,
+		reverseTransition,
+		steps,
+		...rest
+	}) => {
+
+		delete rest.componentRef;
+		delete rest.onNextClick;
+		delete rest.onPrevClick;
+		delete rest.totalIndex;
+
+		return (
+			<div {...rest}>
+				{pageIndicatorType === 'dot' ? steps : null}
+				<Column style={{overflow: 'hidden'}}>
+					<Row style={{height: '100%'}}>
+						{pageIndicatorType === 'dot' ? prevNavigationButton : null}
+						<Cell
+							arranger={arranger}
+							component={ViewManager}
+							duration={400}
+							id="PageViewsContent"
+							index={index}
+							noAnimation={noAnimation}
+							onTransition={onTransition}
+							onWillTransition={onWillTransition}
+							reverseTransition={reverseTransition}
+							style={{overflow: 'hidden'}}
+						>
+							{children}
+						</Cell>
+						{pageIndicatorType === 'dot' ? nextNavigationButton : null}
+					</Row>
+				</Column>
+				{pageIndicatorType === 'number' ? steps : null}
+			</div>
+		);
+	}
+});
+
+const PageViewsDecorator = compose(
+	Changeable({prop: 'index'}),
+	SpotlightContainerDecorator({
+		continue5WayHold: true,
+		defaultElement: [`.${spotlightDefaultClass}`, `#PageViewsContent *`, '#NextNavButton', '#PrevNavButton'],
```

Comment:
Is it safe to using 'id' for searching navigable targets?

Reply:
I fixed to use class for searching navigable targets.